### ISSUE TITLE
[Bug]: The tray icon looks ugly, low-resolution

### ISSUE THREAD

**Issue by paride** [snapcrafters/signal-desktop#127]
### What happened?

On Ubuntu 23.04 (Lunar Lobster, GNOME 44) the signal-desktop icon looks ugly:

![image](https://user-images.githubusercontent.com/7614669/232460876-3f81a3a4-5d4d-46cc-b436-1425cc4116a7.png)

I'm not sure on how it looks on older releases, however I remember it looking good up to a few months ago.

### What should have happened?

The icon should look good :-)

### Output of `snap info $snap_name`

```shell
name:      signal-desktop
summary:   Private messaging from your desktop.
publisher: Snapcrafters
store-url: https://snapcraft.io/signal-desktop
contact:   https://github.com/snapcrafters/signal-desktop/issues
license:   AGPL-3.0-only
description: |
  Private messaging from your desktop.
  
  To use the Signal desktop app, Signal must first be installed on your phone.
  
  **Are you having issues?**
  
  Let us know by creating a new issue here:
  https://github.com/snapcrafters/signal-desktop/issues
  
  **Authors**
  
  This snap is maintained by the Snapcrafters community, and is not necessarily endorsed or
  officially maintained by the upstream developers.
commands:
  - signal-desktop
snap-id:      r4LxMVp7zWramXsJQAKdamxy6TAWlaDD
tracking:     latest/stable
refresh-date: 2 days ago, at 23:55 BST
channels:
  latest/stable:    6.14.0 2023-04-14 (491) 176MB -
  latest/candidate: 6.14.0 2023-04-13 (491) 176MB -
  latest/beta:      ↑                             
  latest/edge:      ↑                             
installed:          6.14.0            (491) 176MB -
```
```


### Output of `snap connections $snap_name`

```shell
Interface               Plug                                   Slot                            Notes
audio-playback          signal-desktop:audio-playback          :audio-playback                 -
audio-record            signal-desktop:audio-record            :audio-record                   -
browser-support         signal-desktop:browser-support         :browser-support                -
camera                  signal-desktop:camera                  :camera                         -
content[gnome-42-2204]  signal-desktop:gnome-42-2204           gnome-42-2204:gnome-42-2204     -
content[gtk-3-themes]   signal-desktop:gtk-3-themes            gtk-common-themes:gtk-3-themes  -
content[icon-themes]    signal-desktop:icon-themes             gtk-common-themes:icon-themes   -
content[sound-themes]   signal-desktop:sound-themes            gtk-common-themes:sound-themes  -
desktop                 signal-desktop:desktop                 :desktop                        -
desktop-legacy          signal-desktop:desktop-legacy          :desktop-legacy                 -
gsettings               signal-desktop:gsettings               :gsettings                      -
home                    signal-desktop:home                    :home                           -
network                 signal-desktop:network                 :network                        -
opengl                  signal-desktop:opengl                  :opengl                         -
removable-media         signal-desktop:removable-media         -                               -
screen-inhibit-control  signal-desktop:screen-inhibit-control  :screen-inhibit-control         -
unity7                  signal-desktop:unity7                  :unity7                         -
wayland                 signal-desktop:wayland                 :wayland                        -
x11                     signal-desktop:x11                     :x11                            -
```
```


### Output of `snap version`

```shell
snap    2.59.1+23.04ubuntu1
snapd   2.59.1+23.04ubuntu1
series  16
ubuntu  23.04
kernel  6.2.0-20-generic
```
```


### Relevant log output

_No response_

### Teminal output of app

_No response_

**Comment by FiSHYsolutions** [comment]
instead of complaining about an ugly tray icon you should be happy having one. users of ubuntu 22.04 doesn't have it for over six month now. 22.04 is a lts release. "lts" means long term support, but nobody cares about it. just give it up. there is no hope someone is willing to fix the ugly one, because nobody is fixing the missing one.

https://github.com/snapcrafters/signal-desktop/issues/87
https://github.com/signalapp/Signal-Desktop/issues/6243
https://github.com/signalapp/Signal-Desktop/issues/6259
https://github.com/signalapp/Signal-Desktop/issues/6352

**Comment by smarek** [comment]
Have you tried libayatana? see https://github.com/signalapp/Signal-Desktop/issues/6243#issuecomment-1574063892

TL;DR: Install libayatana (and all possible sub-packages), restart your WM and signal-desktop, observe tray icon working

**Comment by paride** [comment]
The tray icon is still kind of ugly but looking better now (see attachment). Looks like the icon is not meant to be resized so small. Maybe this issue can be considered solved? This is without libayatana.

![image](https://github.com/snapcrafters/signal-desktop/assets/7614669/4be64977-e813-43ac-a7da-a31d9631485c)

**Comment by smarek** [comment]
@paride i think it still depends on what window-manager(wm)/display-manager(dm), whether your desktop is GTK-2, GTK-3, GTK-4 or differen, and what tray provider you are using.

can you, for my curiosity, but it might help others to solve this, show output of these?

1) `dpkg -l | grep ayatana`
2) `dpkg -l | grep -i indicator | grep -i app`
3) `dpkg -l | grep -i tray`

for me it yields

1)
```
# dpkg -l | grep -i ayatana
ii  gir1.2-ayatanaappindicator-0.1             0.5.92-1                                    amd64        Typelib files for libayatana-appindicator1 (GTK-2+ version)
ii  gir1.2-ayatanaappindicator3-0.1            0.5.92-1                                    amd64        Typelib files for libayatana-appindicator3-1 (GTK-3+ version)
ii  gir1.2-ayatanacommon-0.0                   0.9.8-1                                     amd64        Ayatana System Indicators' common API functions (GObject Introspection)
ii  gir1.2-ayatanaido3-0.4:amd64               0.9.3-1                                     amd64        Widgets and other objects used for Ayatana Indicators - Typelib file
ii  libayatana-appindicator-dev                0.5.92-1                                    amd64        Ayatana Application Indicators (development files, GTK-2+ version)
ii  libayatana-appindicator-doc                0.5.92-1                                    all          Ayatana Application Indicators (documentation files, GTK-2+ version)
ii  libayatana-appindicator1                   0.5.92-1                                    amd64        Ayatana Application Indicators (GTK-2+ version)
ii  libayatana-appindicator3-1                 0.5.92-1                                    amd64        Ayatana Application Indicators (GTK-3+ version)
ii  libayatana-appindicator3-dev               0.5.92-1                                    amd64        Ayatana Application Indicators (development files, GTK-3+ version)
ii  libayatana-common-dev                      0.9.8-1                                     amd64        Ayatana System Indicators' common API functions (development headers)
ii  libayatana-common0:amd64                   0.9.8-1                                     amd64        Ayatana System Indicators' common API functions (shared library)
ii  libayatana-ido3-0.4-0:amd64                0.9.3-1                                     amd64        Widgets and other objects used for Ayatana Indicators
ii  libayatana-ido3-dev                        0.9.3-1                                     amd64        Widgets and other objects used for Ayatana Indicators - development files
ii  libayatana-indicator-dev                   0.9.3-1                                     amd64        panel indicator applet - library development files (GTK-2+)
ii  libayatana-indicator3-7:amd64              0.9.3-1                                     amd64        panel indicator applet - shared library (GTK-3+ variant)
ii  libayatana-indicator3-dev                  0.9.3-1                                     amd64        panel indicator applet - library development files (GTK-3+)
ii  libayatana-indicator3-tools                0.9.3-1                                     amd64        Tools for libayatana-indicator
ii  libayatana-indicator7:amd64                0.9.3-1                                     amd64        panel indicator applet - shared library (GTK-2+ variant)
```

2) 
```
# dpkg -l | grep -i indicator | grep -i app
ii  gir1.2-ayatanaappindicator-0.1             0.5.92-1                                    amd64        Typelib files for libayatana-appindicator1 (GTK-2+ version)
ii  gir1.2-ayatanaappindicator3-0.1            0.5.92-1                                    amd64        Typelib files for libayatana-appindicator3-1 (GTK-3+ version)
ii  libayatana-appindicator-dev                0.5.92-1                                    amd64        Ayatana Application Indicators (development files, GTK-2+ version)
ii  libayatana-appindicator-doc                0.5.92-1                                    all          Ayatana Application Indicators (documentation files, GTK-2+ version)
ii  libayatana-appindicator1                   0.5.92-1                                    amd64        Ayatana Application Indicators (GTK-2+ version)
ii  libayatana-appindicator3-1                 0.5.92-1                                    amd64        Ayatana Application Indicators (GTK-3+ version)
ii  libayatana-appindicator3-dev               0.5.92-1                                    amd64        Ayatana Application Indicators (development files, GTK-3+ version)
ii  libayatana-indicator-dev                   0.9.3-1                                     amd64        panel indicator applet - library development files (GTK-2+)
ii  libayatana-indicator3-7:amd64              0.9.3-1                                     amd64        panel indicator applet - shared library (GTK-3+ variant)
ii  libayatana-indicator3-dev                  0.9.3-1                                     amd64        panel indicator applet - library development files (GTK-3+)
ii  libayatana-indicator7:amd64                0.9.3-1                                     amd64        panel indicator applet - shared library (GTK-2+ variant)
ii  libindicator7:amd64                        0.5.0-4                                     amd64        panel indicator applet - shared library
```

3) 
```
# dpkg -l | grep -i tray
ii  pasystray                                  0.8.1-1                                     amd64        PulseAudio controller for the system tray
ii  xfce4-sntray-plugin                        0.4.13.1-2+b1                               amd64        plugin for xfce4-panel to show StatusNotifierItems - Xfce
ii  xfce4-sntray-plugin-common                 0.4.13.1-2                                  all          plugin for xfce4-panel to show StatusNotifierItems - commoner
```

**Comment by paride** [comment]
I am using a pretty much standard Ubuntu system (running Mantic, the devel release).

```
$ dpkg -l | grep ayatana
ii  gir1.2-ayatanaappindicator3-0.1                  0.5.92-1                                   amd64        Typelib files for libayatana-appindicator3-1 (GTK-3+ version)
ii  gir1.2-ayatanaido3-0.4:amd64                     0.9.3-1                                    amd64        Widgets and other objects used for Ayatana Indicators - Typelib file
ii  libayatana-appindicator3-1                       0.5.92-1                                   amd64        Ayatana Application Indicators (GTK-3+ version)
ii  libayatana-appindicator3-dev                     0.5.92-1                                   amd64        Ayatana Application Indicators (development files, GTK-3+ version)
ii  libayatana-ido3-0.4-0:amd64                      0.9.3-1                                    amd64        Widgets and other objects used for Ayatana Indicators
ii  libayatana-ido3-dev                              0.9.3-1                                    amd64        Widgets and other objects used for Ayatana Indicators - development files
ii  libayatana-indicator3-7:amd64                    0.9.3-1                                    amd64        panel indicator applet - shared library (GTK-3+ variant)
ii  libayatana-indicator3-dev                        0.9.3-1                                    amd64        panel indicator applet - library development files (GTK-3+)
```

```
$ dpkg -l | grep -i indicator | grep -i app
ii  gir1.2-ayatanaappindicator3-0.1                  0.5.92-1                                   amd64        Typelib files for libayatana-appindicator3-1 (GTK-3+ version)
ii  gnome-shell-extension-appindicator               53-1                                       all          AppIndicator, KStatusNotifierItem and tray support for GNOME Shell
ii  libayatana-appindicator3-1                       0.5.92-1                                   amd64        Ayatana Application Indicators (GTK-3+ version)
ii  libayatana-appindicator3-dev                     0.5.92-1                                   amd64        Ayatana Application Indicators (development files, GTK-3+ version)
ii  libayatana-indicator3-7:amd64                    0.9.3-1                                    amd64        panel indicator applet - shared library (GTK-3+ variant)
ii  libayatana-indicator3-dev                        0.9.3-1                                    amd64        panel indicator applet - library development files (GTK-3+)
```

```
$ dpkg -l | grep -i tray
ii  gnome-shell-extension-appindicator               53-1                                       all          AppIndicator, KStatusNotifierItem and tray support for GNOME Shell
```

**Comment by smarek** [comment]
thank you

I'd go with filing bug report against gnome-shell-extension-appindicator 
however i found one report, that might be relevant to you https://github.com/ubuntu/gnome-shell-extension-appindicator/issues/399#issuecomment-1463519333

**Comment by smarek** [comment]
yeah, reading up more on the appindicator extension, its hot mess, https://github.com/ubuntu/gnome-shell-extension-appindicator/issues/319

**Comment by FiSHYsolutions** [comment]
still no tray icon in signal desktop v6.26.0 (snap) using xubuntu 22.04.2 lts

**Comment by merlijn-sebrechts** [comment]
This is my tray icon now: 
![image](https://github.com/snapcrafters/signal-desktop/assets/1492981/018acf80-7e67-4d47-9992-ab6e6bfb8277)


So the issue seems to be fixed now. I'm closing this issue. Please comment if this isn't fixed, and explain which specific linux distro, version and desktop environment you're using.

**Comment by FiSHYsolutions** [comment]
still no tray icon in signal desktop v6.38.0 (snap) using xubuntu 22.04.3 lts (Xfce 4.16)
